### PR TITLE
Use python `CommonData`

### PR DESCRIPTION
**Pull request by scarlehoff**
There has been a half-finished, half-used `python` version of the `CommonData` object for a while https://github.com/NNPDF/nnpdf/blob/9d57bb0df15b7269d00d7d56ad3cba63b966c9b0/validphys2/src/validphys/coredata.py#L202

This PR completes so that it can be used everywhere.

This still doesn't allow to remove `libNNPDF` because the generation of closure test data is still done from `libNNPDF`. Is there someone in Cambridge dealing with that at the moment @Zaharid ?

If not I could also have a go at that after this is merged.

Note that I have not attempted to fix any possible design problems there could be with the python version of the `CommonData` or with how the `libNNPDF` was used. This is just a conversion so that when we move to the new commondata I only have to port one type of `CommonData` instead of two.

#### TODO

- [x] Check all validphys runcards in the example folder to check that I have not missed some corner action not covered by tests.

### CODE REVIEW

**Comment by Zaharid** [comment]
We are going to have a pr for the closure test data soon.

**Comment by scarlehoff** [comment]
Good. I guess is this branch? https://github.com/NNPDF/nnpdf/compare/python_closure_sampling

If so there seem to be no overlap with the changes of this PR so when both are merged it should be possible to remove libNNPDF at last.

**Comment by Zaharid** [comment]
Cc @comane

**Comment by Zaharid** [comment]
Would be good if the export functions were more layered, providing ways to dump the individual files to any streams, mirroring what we have in commondataparser. Among other things, this help us to test easily that writing and reading  roundtrip. These lower level things might go in commondata parser.

**Comment by scarlehoff** [comment]
I didn't check whether they were the same but are they not included in the #1651? Or those are yet something different which is closure test only?

**Comment by Zaharid** [comment]
I guess that part does overlap.

**Comment by scarlehoff** [comment]
Yes, but it is not destructive overlap. When that one is merged the `export` can be changed to use the two functions and done!

**Comment by scarlehoff** [comment]
This is now ready for review. As far as I can test there's nothing broken.

Added @andreab1997 as a reviewer in case this touches the covmat in some way. I think it shouldn't because the thcovmat was freed from `libNNPDF`, but just in case.

**Comment by scarlehoff** [comment]
@Zaharid did you have a look at this?

@andreab1997 please, have a look as well

**Comment by scarlehoff** [comment]
With #1660 it should be possible to merge this to master?

**Comment by scarlehoff** [comment]
I'd like to merge this by the end of the week if there are no blocking issues with it.

I'd create a tag before the merge as the last version that still uses `libNNPDF` in the python code since I expect we will be able to remove all `libNNPDF` imports from `validphys` after this one and #1651 (or #1660) are merged.

File: validphys2/src/validphys/coredata.py
Comment (AleCandido) [inline validphys2/src/validphys/coredata.py:384]: Same note about the encoding.

File: validphys2/src/validphys/filters.py
Comment (AleCandido) [inline validphys2/src/validphys/filters.py:18]: I guess the plural would be more consistent with the content of `commondataparser.py` (btw it is also not separating by underscore - both fine, but I would pick just one).

File: validphys2/src/validphys/filters.py
Comment (AleCandido) [inline validphys2/src/validphys/filters.py:519]: Sphinx is inferring the `:py:class:` part here, if you wish you can simplify with
```suggestion
    commondata: validphys.coredata.CommonData
```

File: validphys2/src/validphys/core.py
Comment (scarlehoff) [reply validphys2/src/validphys/core.py:440]: Yes. I think the last dependency will be dropped by #1651, but now after applying the changes I'll check whether a `load` is left anywhere else. Maybe it can be removed also from here.

File: validphys2/src/validphys/core.py
Comment (scarlehoff) [reply validphys2/src/validphys/core.py:478]: Oh, damn, this PR was in the freezer for so long that I forgot to remove them. Thanks for noticing :__
```suggestion
                cd = cd.with_cuts(intmask)
```

File: validphys2/src/validphys/coredata.py
Comment (scarlehoff) [reply validphys2/src/validphys/coredata.py:292]: Both this (and the one below) are left as a method because they were used for the old commondata as methods (with those names).

I prefer to leave them like that while the `LegacyCommonData` still exist.

File: validphys2/src/validphys/coredata.py
Comment (scarlehoff) [reply validphys2/src/validphys/coredata.py:367]: This is the one I was discussing with @Zaharid during the code meeting. This function will be removed by #1660 or #1651

(but yes, the name is for compatibility with the old one, it does exactly the same under the same method)

**Comment by scarlehoff** [comment]
RE `LegacyCommonData`, while the fit is not using it, there are too many `load` scattered around the code. Once the pseudodata one is merged and I'm sure that no libNNPDF _should_ be use I'll remove all of them.

If then there are functions that are just there and not being used (it wouldn't be the first time) I'll just remove them.

File: validphys2/src/validphys/commondataparser.py
Comment (Zaharid) [inline validphys2/src/validphys/commondataparser.py:124]: These should be frozen by default. We may want to be caching based on this a.t some point

Comment (scarlehoff) [reply validphys2/src/validphys/commondataparser.py:124]:
```suggestion
@dataclasses.dataclass(frozen=True)
```

**Comment by Zaharid** [comment]
I am inclined to merge this as is, provided various tests pass.

File: validphys2/src/validphys/commondataparser.py
Comment (RoyStegeman) [inline validphys2/src/validphys/commondataparser.py:5]:
```suggestion
interfaced with common Python libraries. 
```

File: validphys2/src/validphys/dataplots.py
Comment (RoyStegeman) [inline validphys2/src/validphys/dataplots.py:967]: why not do `KINNAMES[0]"? Hardcoding it in one place but sharing/using KINNAMES in another place is a bit confusing.

Comment (AleCandido) [reply validphys2/src/validphys/dataplots.py:967]: I agree, and hard-coded strings make it also harder to refactor. I plainly missed this.

File: validphys2/src/validphys/coredata.py
Comment (scarlehoff) [reply validphys2/src/validphys/coredata.py:367]: it is already removed in a different PR, don't want to have to update #1660  because there is a conflict :P

File: validphys2/src/validphys/dataplots.py
Comment (scarlehoff) [reply validphys2/src/validphys/dataplots.py:967]: > why not do `KINNAMES[0]"?

Because I wasn't really looking at the strings but rather at removing the C++ methods. I've changed this together with the `_` suggestion.

Comment (RoyStegeman) [reply validphys2/src/validphys/dataplots.py]: I'll stop phrasing my comments as questions ;)

**Review by RoyStegeman — APPROVED** [review]

**Review by AleCandido — APPROVED** [review]

**Pull request merged**
